docassemble/ALToolbox/data/questions/al_income_demo.yml

```diff
@@ -4,6 +4,66 @@ metadata:
 include:
   - al_income.yml
 ---
+mandatory: True
+code: |
+  class ALSeasonalItemizedJob(ALItemizedJobList):
+      """
+      Attributes:
+      .to_add {ALItemizedValueDict} Dict of ALItemizedValues that would be added
+          to a job's net total, like wages and tips.
+      .to_subtract {ALItemizedValueDict} Dict of ALItemizedValues that would be
+          subtracted from a net total, like union dues or insurance premiums.
+      .times_per_year {float} A denominator of a year, like 12 for monthly, that
+          represents how frequently the income is earned
+      .employer {Individual} (Optional) Individual assumed to have a name and,
+          optionally, an address and phone.
+      .source {str} (Optional) The category of this item, like "public service".
+      """
+      def init(self, *pargs, **kwargs):
+          super().init(*pargs, **kwargs)
+          self.ask_number = True
+          self.target_number = 12
+          month_names = [
+              "january", "february", "march", "april", "may", "june",
+              "july", "august", "september", "october", "november"
+          ]
```

Comment:
Same question as above. I'm just curious if this needs to be limited to chunking by month. Biweekly pay is a bit more common for most folks.

Reply:
That's a good question. @CaroRob and I have been thinking about that discussion. I had hesitated to go beyond months since some seemed reluctant about getting even this granular, but if folks are open to that discussion, I am too.

Reply:
Think about whether limiting it to months helps achieve any of the goals. It might, but I'm not sure. I think you discussed letting people avoid math. Monthly chunking doesn't seem like it would save any math.

Maybe letting users enter a date range? E.g., teachers usually have 9 month contracts, although they don't always have the choice to get paid over 9 months rather than 12.

Reply:
I've made it more flexible. Will wait to fully adjust the demo till I get the go-ahead. Here's a sample file that shows the most basic application (you must answer that, yes, your job is seasonal). It's .txt and you have to have to change it to .yml because GitHub won't let me upload a .yml.

[seasonal_income_demo.txt](https://github.com/SuffolkLITLab/docassemble-ALToolbox/files/13462189/seasonal_income_demo.txt)